[JSC] Fix WebAssembly.Tag.getArg's thrown error type

Constellation opened the pull request:
#### b1248207241a66f6361f4a2f035f18cc7caed4ae
<pre>
[JSC] Fix WebAssembly.Tag.getArg&apos;s thrown error type
<a href="https://bugs.webkit.org/show_bug.cgi?id=247223">https://bugs.webkit.org/show_bug.cgi?id=247223</a>
rdar://101705666

Reviewed by Alexey Shvayka.

1. We should use toNonWrappingUint32 to get Uint32 for getArg index
2. We should throw RangeError instead of TypeError for out of bounds getArg

<a href="https://webassembly.github.io/exception-handling/js-api/#dom-exception-getarg">https://webassembly.github.io/exception-handling/js-api/#dom-exception-getarg</a>

* LayoutTests/imported/w3c/web-platform-tests/wasm/compile_worker.js:
* LayoutTests/imported/w3c/web-platform-tests/wasm/create_multiple_memory.worker.js:
* LayoutTests/imported/w3c/web-platform-tests/wasm/jsapi/exception/getArg.tentative.any.js:
(test):
* LayoutTests/imported/w3c/web-platform-tests/wasm/resources/load_wasm.js:
* LayoutTests/imported/w3c/web-platform-tests/wasm/webapi/origin.sub.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/wasm/webapi/origin.sub.any.js:
* LayoutTests/imported/w3c/web-platform-tests/wasm/webapi/origin.sub.any.worker-expected.txt:
* Source/JavaScriptCore/wasm/js/WebAssemblyExceptionPrototype.cpp:
(JSC::JSC_DEFINE_HOST_FUNCTION):
(JSC::getTag): Deleted.

Canonical link: <a href="https://commits.webkit.org/256158@main">https://commits.webkit.org/256158@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7723621de72408eaa32e0905e89303aa2bfba651

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/94938 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/77/builds/4086 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/27844 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/104543 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/164807 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/4171 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/32269 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/87244 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/100463 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/100607 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/3014 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/81417 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/87244 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/84928 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/27844 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/87244 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/1/builds/86077 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/66/builds/38678 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/27844 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/81303 "Built successfully") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/67/builds/36509 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/27844 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/28003 "Passed tests") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/63/builds/40435 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/81417 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/83976 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2033 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/42411 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/27844 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/18980 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

shvaikalesh reviewed (approved):
Nice! A spec link in the ChangeLog would a great addition though: https://webassembly.github.io/exception-handling/js-api/#dom-exception-getarg.

Constellation committed:
[JSC] Fix WebAssembly.Tag.getArg's thrown error type
https://bugs.webkit.org/show_bug.cgi?id=247223
rdar://101705666

Reviewed by Alexey Shvayka.

1. We should use toNonWrappingUint32 to get Uint32 for getArg index
2. We should throw RangeError instead of TypeError for out of bounds getArg

https://webassembly.github.io/exception-handling/js-api/#dom-exception-getarg

* LayoutTests/imported/w3c/web-platform-tests/wasm/compile_worker.js:
* LayoutTests/imported/w3c/web-platform-tests/wasm/create_multiple_memory.worker.js:
* LayoutTests/imported/w3c/web-platform-tests/wasm/jsapi/exception/getArg.tentative.any.js:
(test):
* LayoutTests/imported/w3c/web-platform-tests/wasm/resources/load_wasm.js:
* LayoutTests/imported/w3c/web-platform-tests/wasm/webapi/origin.sub.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/wasm/webapi/origin.sub.any.js:
* LayoutTests/imported/w3c/web-platform-tests/wasm/webapi/origin.sub.any.worker-expected.txt:
* Source/JavaScriptCore/wasm/js/WebAssemblyExceptionPrototype.cpp:
(JSC::JSC_DEFINE_HOST_FUNCTION):
(JSC::getTag): Deleted.

Canonical link: https://commits.webkit.org/256158@main
